capa/rules.py

```diff
@@ -277,6 +277,42 @@ def parse_description(s, value_type, description=None):
     return value, description
 
 
+def parse_meta(meta):
+    """
+    further process meta items such as MBC
+    """
+    mbcs = meta.get("mbc", [])
+    if not isinstance(mbcs, list):
+        raise InvalidRule("MBC mapping must be a list")
+
+    if mbcs:
+        meta["mbc"] = [parse_canonical_mbc(mbc) for mbc in mbcs]
+
+    # TODO att&ck
+
+    return meta
+
+
+def parse_canonical_mbc(mbc):
+    """
+    parse capa's canonical MBC representation: `OBJECTIVE::Behavior::Method [Identifier]`
+    """
+    objective, _, rest = mbc.partition("::")
+    if "::" in rest:
+        behavior, _, rest = rest.partition("::")
+        method, _, id = rest.rpartition(" ")
+    else:
+        behavior, _, id = rest.rpartition(" ")
+        method = ""
+    return {
+        "id": id.lstrip("[").rstrip("]"),
+        "objective": objective,
+        "behavior": behavior,
+        "method": method,
+        # TODO "micro-behavior": "",
+    }
```

Comment:
im concerned here that we expect exactly two or three components and never any further subscope. we should parse and expose all the components, even if there are seven. so perhaps something like:

```suggestion
    path = mbc.split("::")
    if len(path) > 0:
        last = path.pop()
        last, _, id = last.partition(" ")
        id = id.lstrip("[").rstrip("]")
        path.append(last)
    
    if len(path) > 0:
        objective = path[0]
    if len(path) > 1:
        behavior = path[1]
    if len(path) > 2:
        method = path[2]

    return {
        "path": path,
        "id": id or "",
        "objective": objective or "",
        "behavior": behavior or "",
        "method": method or "",
    }
```